samcli/lib/build/app_builder.py

```diff
@@ -663,3 +706,89 @@ def _parse_builder_response(stdout_data: str, image_name: str) -> Dict:
+
+        """
+
+        name = function.name
+        result = {}
+
+        if file_env_vars and inline_env_vars:
+            for file_env_var in file_env_vars.values():
+                if not isinstance(file_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            for inline_env_var in inline_env_vars.values():
+                if not isinstance(inline_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            LOG.debug("Environment variables data is being parsed")
+            # CloudFormation parameter file format
+            parameter_result = file_env_vars.get("Parameters", {})
+            result = parameter_result.copy()
+            inline_parameter_result = inline_env_vars.get("Parameters", {})
+            result.update(inline_parameter_result)
+            specific_result = file_env_vars.get(name, {})
+            result.update(specific_result)
+            inline_specific_result = inline_env_vars.get(name, {})
+            result.update(inline_specific_result)
+
+        elif file_env_vars:
+            for file_env_var in file_env_vars.values():
+                if not isinstance(file_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            LOG.debug("Environment variables data is being parsed")
+            # CloudFormation parameter file format
+            parameter_result = file_env_vars.get("Parameters", {})
+            result = parameter_result.copy()
+            specific_result = file_env_vars.get(name, {})
+            result.update(specific_result)
+
+        elif inline_env_vars:
+            for inline_env_var in inline_env_vars.values():
+                if not isinstance(inline_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            LOG.debug("Environment variables data is being parsed")
+            # CloudFormation parameter file format
+            parameter_result = inline_env_vars.get("Parameters", {})
+            result = parameter_result.copy()
+            specific_result = inline_env_vars.get(name, {})
+            result.update(specific_result)
```

Comment:
this `if ... elif ... elif` can be simplified to avoid code duplication. 

You can initialize a empty `parameter_result` and go through `file_env_vars` and `inline_env_vars` if they exist.

Reply:
I don't really believe so, the priority needs to be written this way to be maintained, I don't see any ways to simplify this

Reply:
This?

```py
@staticmethod
def _make_env_vars(function: Function, file_env_vars: Dict, inline_env_vars: Optional[Dict]) -> Dict:
    """Returns the environment variables configuration for this function
    Parameters
    ----------
    function : samcli.lib.providers.provider.Function
        Lambda function to generate the configuration for
    Returns
    -------
    dictionary
        Environment variable configuration for this function
    Raises
    ------
    samcli.commands.local.lib.exceptions.OverridesNotWellDefinedError
        If the environment dict is in the wrong format to process environment vars
    """

    name = function.name
    result = {}


    # validate and raise OverridesNotWellDefinedError
    for env_var in (file_env_vars or {}).values() + (inline_env_vars or {}).values():
        if not isinstance(env_var, dict):
            reason = """
                        Environment variables in incorrect format
                        """
            LOG.debug(reason)
            raise OverridesNotWellDefinedError(reason)

    if file_env_vars:
        parameter_result = file_env_vars.get("Parameters", {})
        result.update(parameter_result)
    
    if inline_env_vars:
        inline_parameter_result = inline_env_vars.get("Parameters", {})
        result.update(inline_parameter_result)
    
    if file_env_vars:
        specific_result = file_env_vars.get(name, {})
        result.update(specific_result)
    
    if inline_env_vars:
        inline_specific_result = inline_env_vars.get(name, {})
        result.update(inline_specific_result)

    return result
```

Reply:
sure I'm not sure if this would create any behaviour but thanks let me try to run it with this!

Reply:
This proposed version only needs less than 30 lines of code and with better readability.

Reply:
yeah, you have unit tests to verify it.